core/state_processor.go

```diff
@@ -1812,7 +1910,11 @@ func ProcessQiTx(tx *types.Transaction, chain ChainContext, checkSig bool, isFir
 		if ETXPGas > *etxPLimit {
 			return nil, nil, nil, fmt.Errorf("tx [%v] emits too many cross-prime ETXs for block. gas emitted: %d, gas limit: %d", tx.Hash().Hex(), ETXPGas, etxPLimit), nil
 		}
-		etxInner := types.ExternalTx{Value: totalConvertQitOut, To: &convertAddress, Sender: common.ZeroAddress(location), EtxType: uint64(etxType), OriginatingTxHash: tx.Hash(), Gas: 0, Data: data} // Value is in Qits not Denomination
+		// gas left
+		gasLeft := new(big.Int).Sub(txFeeInQuai, minimumFeeInQuai)
```

Comment:
What is gas used for?

Reply:
for refund

Reply:
It's not though, I just checked. I don't think we need this really

Reply:
1. yeah thats right, since we have done this for all etxs, thought is the correct thing to do here